### PR TITLE
DOC: add examples to build_block_contiguity

### PR DESCRIPTION
**Pull request by dancejod**
Added Examples section to `build_block_contiguity` docstring

### CODE REVIEW

**Comment by jGaboardi** [comment]
@dancejod Thanks for the PR! Can you run `ruff check libpysal` and `ruff format libpysal` locally to fix the linting and formatting errors? After that please repush.

**Review by jGaboardi — APPROVED** [review]

File: libpysal/graph/base.py [hunk 1]
```diff
@@ -359,6 +359,35 @@ def build_block_contiguity(cls, regimes):
         -------
         Graph
             libpysal.graph.Graph encoding block contiguity
+
+        Examples
+        -------
+        >>> import geopandas as gpd
+        >>> from geodatasets import get_path
+        >>> france = gpd.read_file(get_path('geoda guerry')).set_index('Dprmnt')
```
Comment (martinfleis) [inline libpysal/graph/base.py:367]:
```suggestion
        >>> france = gpd.read_file(get_path('geoda guerry')).set_index('Dprmnt')
        
        In the GeoDa Guerry dataset, the Region column reflects the region 
        (North, East, West, South or Central) to which each department belongs.
        
```

File: libpysal/graph/base.py [hunk 2]
```diff
@@ -359,6 +359,35 @@ def build_block_contiguity(cls, regimes):
...
+        >>> france[['Region', 'geometry']].head()
+                     Region                                           geometry
+        Dprtmnt
+        Ain               E  POLYGON ((801150.000 2092615.000, 800669.000 2...
+        Aisne             N  POLYGON ((729326.000 2521619.000, 729320.000 2...
+        Allier            C  POLYGON ((710830.000 2137350.000, 711746.000 2...
+        Basses-Alpes      E  POLYGON ((882701.000 1920024.000, 882408.000 1...
+        Hautes-Alpes      E  POLYGON ((886504.000 1922890.000, 885733.000 1...
+        >>> block_contiguity = graph.Graph.build_block_contiguity(france['Region'])
```
Comment (martinfleis) [inline libpysal/graph/base.py:376]:
```suggestion
        
        Using the ``"Region"`` labels as ``regimes`` then identifies all departments 
        within the region as neighbors.

        >>> block_contiguity = graph.Graph.build_block_contiguity(france['Region'])
```

File: libpysal/graph/base.py [hunk 3]
```diff
@@ -359,6 +359,35 @@ def build_block_contiguity(cls, regimes):
         -------
         Graph
             libpysal.graph.Graph encoding block contiguity
+
+        Examples
+        -------
```
Comment (martinfleis) [inline libpysal/graph/base.py:364]:
```suggestion
        --------
```

**Review by martinfleis — COMMENTED** [review]
Some suggestions.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**